docs/guides/expectations/how_to_create_and_edit_expectations_with_a_profiler.md

```diff
@@ -9,10 +9,230 @@ This guide will help you create a new Expectation Suite by profiling your data w
+    ignored_columns=ignored_columns,
+    not_null_only=not_null_only,
+    table_expectations_only=table_expectations_only,
+    value_set_threshold=value_set_threshold)
+
+suite = profiler.build_suite()
+
+```
+
+**Once you have instantiated a profiler with parameters specified, you must re-instantiate the profiler if you wish to change any of the parameters.**
+
+### Semantic Types Dictionary Configuration
+
+The profiler is fairly rudimentary - if it detects that a column is numeric, it will create numeric expectations (e.g. ``expect_column_mean_to_be_between``). But if you are storing foreign keys or primary keys as integers, then you may not want numeric expectations on these columns. This is where the semantic_types dictionary comes in.
+
+The available semantic types that can be specified in the UserConfigurableProfiler are "numeric", "value_set", and "datetime". The expectations created for each of these types is below. You can pass in a dictionary where the keys are the semantic types, and the values are lists of columns of those semantic types.
+
+When you pass in a `semantic_types_dict`, the profiler will still create table-level expectations, and will create certain expectations for all columns (around nullity and column proportions of unique values). It will then only create semantic-type-specific expectations for those columns specified in the semantic_types dict.
+
+```python
+semantic_types_dict = {
+    "numeric": ["acctbal"],
+    "value_set": ["nationkey","mktsegment", 'custkey', 'name', 'address', 'phone', "acctbal"]
+}
+
+validator = context.get_validator(
+    batch_request=BatchRequest(**batch_request),
+    expectation_suite_name=expectation_suite_name
+)
+
+profiler = UserConfigurableProfiler(
+    profile_dataset=validator,
+    semantic_types_dict=semantic_types_dict
+)
+suite = profiler.build_suite()
+```
+
+These are the expectations added when using a `semantics_type_dict`:
+
+**Table expectations:**
+- `expect_table_row_count_to_be_between`
+- `expect_table_columns_to_match_ordered_list`
+
+
+**Expectations added for all included columns**
+- `expect_column_value_to_not_be_null` (if a column consists of more than 50% null values, this will instead add `expect_column_values_to_be_null`)
+- `expect_column_proportion_of_unique_values_to_be_between`
+- `expect_column_values_to_be_in_type_list`
+
+
+**Value set expectations**
+- `expect_column_values_to_be_in_set`
+
+
+**Datetime expectations**
+- `expect_column_values_to_be_between`
+
+
+**Numeric expectations**
+- `expect_column_min_to_be_between`
+- `expect_column_max_to_be_between`
+- `expect_column_mean_to_be_between`
+- `expect_column_median_to_be_between`
+- `expect_column_quantile_values_to_be_between`
+
+
+**Other expectations**
+- `expect_column_values_to_be_unique` (if a single key is specified for `primary_or_compound_key`)
+- `expect_compound_columns_to_be_unique` (if a compound key is specified for `primary_or_compound_key`)
```

Comment:
Could we link to these? Perhaps the Expectation gallery? Not sure if there's clear documentation but it would be nice to be able to see what these are from this page.

Reply:
☝️ This. I'd love to see the gallery get more traction in our docs.

Reply:
Great idea. I just made the other fixes; I'll add these links next week.